File: DataSources/Amazon-Athena/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs
using Reveal.Sdk;
using Reveal.Sdk.Data;
using Reveal.Sdk.Data.Amazon.Athena;

namespace RevealSdk.Server.Reveal
{
    public class DataSourceProvider : IRVDataSourceProvider
    {
        public async Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId, RVDataSourceItem dataSourceItem)
        {
            await ChangeDataSourceAsync(userContext, dataSourceItem.DataSource);

            if (dataSourceItem is RVAthenaDataSourceItem dsi)
            {
                if (dsi.Id == "my-data-source-item")
                {
                    dsi.Table = "your_table_name";
                }
            }

            return dataSourceItem;
        }

        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext, RVDashboardDataSource dataSource)
        {
            if (dataSource is RVAthenaDataSource databricksDataSource)
            {
                databricksDataSource.Region = "your_region";
                databricksDataSource.Database = "your_database_name";
            }

            return Task.FromResult(dataSource);
        }
    }
}

File: DataSources/Amazon-Athena/server/spring-boot-jersey/src/main/java/com/server/reveal/DataSourceProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVDataSourceProvider;
import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.model.RVDashboardDataSource;
import com.infragistics.reveal.sdk.api.model.RVDataSourceItem;
import com.infragistics.reveal.sdk.api.model.RVAthenaDataSource;
import com.infragistics.reveal.sdk.api.model.RVAthenaDataSourceItem;

public class DataSourceProvider implements IRVDataSourceProvider {
    public RVDataSourceItem changeDataSourceItem(IRVUserContext userContext, String dashboardsID, RVDataSourceItem dataSourceItem) {
        //update underlying data source
        changeDataSource(userContext, dataSourceItem.getDataSource());

        if (dataSourceItem instanceof RVAthenaDataSourceItem dsi) {
            if (dsi.getId() == "my-data-source-item") {
                dsi.setTable("your_table_name");
            }            
        }
        return dataSourceItem;
    }

    public RVDashboardDataSource changeDataSource(IRVUserContext userContext, RVDashboardDataSource dataSource) {

        if (dataSource instanceof RVAthenaDataSource ds) {
            ds.setRegion("your_region");
            ds.setDatabase("your_database_name");
        }
        return dataSource;
    }
}
